circularBuffer.cpp

```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+#include "circularBuffer.hpp"
+
+template <class T> T      circularBuffer<T>::buffer[BufferSize] = {};
+template <class T> size_t circularBuffer<T>::head =                0;
+template <class T> size_t circularBuffer<T>::tail =                0;
+template <class T> bool   circularBuffer<T>::full =            false;
+
+template <class T> void circularBuffer<T>::put_overwrite(T item)
+{
+  buffer[head] = item;
+  if (true == full)
+  {
+    tail = (tail + 1) % BufferSize;
+  }
+  head = (head + 1) % BufferSize;
+  full = (head == tail);
+  return;  
```

Comment:
What's the reason for empty `return`  statements at the end of the function?

Reply:
Thats a matter of style. Some code style guides force the same. 

Anyway this is not wrong and is more alligned to the machine code than leaving it away.

Reply:
Should I change it?

Reply:
No it's not necessary to change, thank you